### PR TITLE
fix(gatsby-source-contentful): LongText fields require markdown transformer

### PR DESCRIPTION
**Pull request by slk333**
**TLDR: The main problem is that if you follow the current documentation, you end up with raw markdown in your pages instead of the expected formatted content**

As shown here: https://github.com/gatsbyjs/gatsby/issues/3205#issuecomment-351603320

> The source plugins still exposes the raw markdown but generally you'd want to use the transformed HTML as in the above code sample.

this means doing the following will return raw markdown instead of html.
```{
body{
  body
  }
}
```
You have to use a transformer plugin to transform the markdown into html.
The doc was misleading since it suggested that using LongText can be used without a transformer, and said that the markdown was transformed by Gatsby (incorrect). As a result, if someone were to follow the example, he would get raw markdown displaying in the webpage, as opposed to the expected result.

### CODE REVIEW

**Comment by slk333** [comment]
Someone suggested that I use react-markdown to easily convert the markdown to react. I guess it's obvious for some people, but for some people it's not.
But using react-markdown still accounts for installing an extra plug-in, so I still think we need to express that in the documentation.

File: packages/gatsby-source-contentful/README.md [hunk 1]
```diff
@@ -237,23 +237,34 @@ You might query for a **single** node inside a component in your `src/components
 
 #### A note about LongText fields
 
-If you include fields with a `LongText` type in your Contentful `ContentType`, their returned value will be **an object not a string**. This is because Contentful LongText fields are Markdown by default. In order to handle the Markdown content properly, this field type is created as a child node so Gatsby can transform it to HTML.
+If you include fields with a `LongText` type in your Contentful `ContentType`, their returned value will be **an object not a string**. This is because Contentful LongText fields are Markdown by default. In order to handle the Markdown content properly, this field type is created as a child node so that a transformer plugin such as gatsby-transformer-remark or react-markdown, can transform it to HTML.
```
Comment (vladar) [inline packages/gatsby-source-contentful/README.md]: Can you explain a bit why you included `react-markdown` here? I afraid mentioning it here could be confusing as it is not a gatsby plugin. I may be missing something of course.

Also, I suggest adding a link to `gatsby-transformer-remark`:

```suggestion
If you include fields with a `LongText` type in your Contentful `ContentType`, their returned value will be **an object not a string**. This is because Contentful LongText fields are Markdown by default. In order to handle the Markdown content properly, this field type is created as a child node so that a transformer plugin such as [gatsby-transformer-remark](https://www.gatsbyjs.org/packages/gatsby-transformer-remark/), can transform it to HTML.
```

**Comment by slk333** [comment]
pinging @jserrao as he authored this part I'm trying to clarify.

**Comment by jserrao** [comment]
@slk333 - I remember when I wrote this, you had to include `body` inside the query or it wouldn't work correctly - honestly can't remember the specifics of whether it returned markdown or HTML. I trust that your improvement is correct slk333.

File: packages/gatsby-source-contentful/README.md (hunk 1 above)
Comment (jserrao) [reply packages/gatsby-source-contentful/README.md]: @vladar - your tweak would make sense, trying to show only the gatsby ecosystem transformer.

**Comment by slk333** [comment]
at the moment, here's what you get when you query the child node
Screenshot 2020-01-09 at 22 40 54

**Comment by jserrao** [comment]
Weird, that night have changed. I don't think it was always encoded like
that.

**Comment by slk333** [comment]
sorry I didn't intend to delay the review, didn't see you asked for a review. Thanks @vladar

**Comment by vladar** [comment]
Looks like `lint` checks are failing. Try running `yarn format` on a repo when you are ready. Also, there is a `WIP` label in case if you still work on this PR.

P.S. Thanks for this fix by the way!

**Comment by slk333** [comment]
Hi @vladar , thanks for your help! I'm still a bit confused on how to fix the lint check failure. Do I have to download the whole gatsby repo and run `yarn format` directly on that repo?

**Comment by laurieontech** [comment]
Was able to run it for you :) Thanks so much for the PR!
